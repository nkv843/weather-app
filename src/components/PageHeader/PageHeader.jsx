import React from 'react';
import PropTypes from 'prop-types';
import classNames from './PageHeader.module.css';

const PageHeader = ({ loading, children }) => (
  <header className={classNames.header}>
    <div className={classNames.container}>
      <div className={classNames.logo} />
      <h1 className={classNames.title}>WTWT in</h1>
      {children}
      {loading
        ? <h1 className={classNames.loading} data-testid="loading">Loading</h1>
        : null}
    </div>
  </header>
);

PageHeader.propTypes = {
  loading: PropTypes.bool,
  children: PropTypes.shape(),
};

PageHeader.defaultProps = {
  loading: false,
  children: null,
};

export default PageHeader;
