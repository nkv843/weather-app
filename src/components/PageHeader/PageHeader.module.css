.container {
  display: flex;
  flex-wrap: wrap;
  background-color: #fff;
  align-items: center;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, .25);
  padding: 20px 30px 0 50px;
  margin-bottom: 40px;

}

.title {
  font-size: 30px;
  margin: 0 20px 20px 0;
  color: rgb(255, 106, 203);
}

.logo {
  min-width: 50px;
  height: 50px;
  margin: 0 20px 20px 0;
  background: url("../../assets/img/favicon.svg");
}

.loading {
  color: rgb(245, 150, 212);
  font-size: 30px;
  margin: 0 20px 20px 0;
  padding-right: 50px;
  background: no-repeat center right;
  background-size: contain;
  background-image: url('../../assets/img/loading.svg');
}
