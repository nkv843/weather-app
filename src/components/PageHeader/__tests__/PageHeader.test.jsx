import { render } from '@testing-library/react';
import React from 'react';
import PageHeader from '../PageHeader';

describe('PageHeader', () => {
  it('should render children', () => {
    const { getByTestId } = render(
      <PageHeader loading={false}>
        <p data-testid="headerChildren" />
      </PageHeader>,
    );
    const children = getByTestId('headerChildren');
    expect(children).toBeInTheDocument();
  });

  it('should hide loading while it is false', () => {
    const { getByTestId } = render(<PageHeader loading={false} />);
    expect(() => getByTestId('loading')).toThrow();
  });

  it('should indicate loading while it is true', () => {
    const { getByTestId } = render(<PageHeader loading />);
    const loadingIndicatior = getByTestId('loading');
    expect(loadingIndicatior).toBeInTheDocument();
  });

  it('should match snapshot with no props', () => {
    const { baseElement } = render(<PageHeader />);
    expect(baseElement).toMatchSnapshot();
  });
});
