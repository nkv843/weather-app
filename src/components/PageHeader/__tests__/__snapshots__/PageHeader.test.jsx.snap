// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should match snapshot with no props 1`] = `
<body>
  <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="logo"
        />
        <h1
          class="title"
        >
          WTWT in
        </h1>
        <div />
      </div>
    </header>
  </div>
</body>
`;
